Making tweaks to allow project to build on readthedocs.org.

dhermes opened the pull request:
In particular
- Adding a requirements.txt file to aid in installing docs-only
  dependencies (this is a feature of readthedocs, see
  https://docs.readthedocs.org/en/latest/builds.html)
- Updating comment before `import django` in docs/conf.py. Without
  the setup that or tox `docs` env provides or the dependencies
  in docs/requirements.txt, sphinx-build will fail due to missing
  `django`
- Adding a section to docs/conf.py which downloads the
  App Engine SDK and adds it to the import path
- Adding base (non-docs) dir so sources get added when generating
  HTML (e.g. sys.path.append('..') within docs/ dir)
- Using package distribution for docs version instead of manually
  importing oauth2client in docs/conf.py.

@nathanielmanistaatgoogle See http://oauth2client.readthedocs.org/en/rtd-test/ for a final view of what this looks like (and compare it to http://google.github.io/oauth2client/)

dhermes commented:
@nathanielmanistaatgoogle Anything holding back a review here?

nathanielmanistaatgoogle commented on docs/requirements.txt:
```diff
@@ -0,0 +1,9 @@
+django
+flask
+keyring
+pycrypto==2.6
```
How much do these have to be ==? Why can't this and the next line be >=?

dhermes replied:
It was copy-pasta from `tox.ini`. Good call.

nathanielmanistaatgoogle commented on docs/conf.py:
```diff
@@ -40,7 +48,24 @@
 
 # We want to set the RTD theme, but not if we're on RTD.
-if os.environ.get('READTHEDOCS', '') != 'True':
+if os.environ.get('READTHEDOCS', '') == 'True':
```
My case-sensitivity-bug spidey-sense is tingling. How important is it that values such as "TRUE", "true", "tru", and "tRuE" cause this expression to evaluate to False?

dhermes replied:
Mine too.

It is what the RTD folks [recommend](https://docs.readthedocs.org/en/latest/faq.html#how-do-i-change-behavior-for-read-the-docs). Honestly I'd rather just do `os.environ.has_key('READTHEDOCS')`.

nathanielmanistaatgoogle replied:
That example uses None as the second argument in the call to os.environ.get, and that's what I prefer as well. Any reason you went with the empty string instead?

dhermes replied:
Craig did it, I just changed the existing stuff in `docs/conf.py`.

I'll do whatever you like, just let me know.

nathanielmanistaatgoogle replied:
None, please. Any other value calls attention to itself and invites the question "now why isn't that None?".

dhermes replied:
Shall I just use `os.getenv`? Also, what value should I compare it to (case-sensitive or not)?

nathanielmanistaatgoogle replied:
Let's follow their example.

Just using os.getenv sounds like a bad idea in case the environment is ever populated with a "READTHEDOCS"/"False" key-value pair.

dhermes replied:
I meant `os.getenv('foo')` is equivalent to `os.environ.get('foo', None)` is equivalent to `os.environ.get('foo')`. Sending now.

nathanielmanistaatgoogle commented:
Thanks for the poke.

dhermes committed:
Making tweaks to allow project to build on readthedocs.org.

In particular

- Adding a requirements.txt file to aid in installing docs-only
  dependencies (this is a feature of readthedocs, see
  https://docs.readthedocs.org/en/latest/builds.html)
- Updating comment before `import django` in docs/conf.py. Without
  the setup that or tox `docs` env provides or the dependencies
  in docs/requirements.txt, sphinx-build will fail due to missing
  `django`
- Adding a section to docs/conf.py which downloads the
  App Engine SDK and adds it to the import path
- Adding base (non-docs) dir so sources get added when generating
  HTML (e.g. sys.path.append('..') within docs/ dir)
- Using package distribution for docs version instead of manually
  importing oauth2client in docs/conf.py.

dhermes commented:
@nathanielmanistaatgoogle Change (from `''` to `None`) sent

nathanielmanistaatgoogle commented:
LGTM and waiting on test results (which I believe will be forthcoming...).

dhermes commented:
Looks like the `docs` env failed to fetch the GAE SDK, I restarted it and amazingly didn't have to restart the others (good job Travis?).

dhermes commented:
Succeeded on rebuild.

dhermes commented:
Victory: http://oauth2client.readthedocs.org
